lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GaussianProcessFitterResult.cxx

```diff
@@ -0,0 +1,235 @@
+  if (size != outputSample.getSize())
+    throw InvalidArgumentException(HERE) << "In GaussianProcessFitterResult::GaussianProcessFitterResult, input & output sample have different size. input sample size = " << size << ", output sample size = " << outputSample.getSize();
+}
+
+
+/* Virtual constructor */
+GaussianProcessFitterResult * GaussianProcessFitterResult::clone() const
+{
+  return new GaussianProcessFitterResult(*this);
+}
+
+
+/* String converter */
+String GaussianProcessFitterResult::__repr__() const
+{
+  return OSS(true) << "class=" << getClassName()
+         << ", covariance model=" << covarianceModel_
+         << ", basis=" << basis_
+         << ", trend coefficients=" << beta_;
+}
+
+String GaussianProcessFitterResult::__str__(const String & offset) const
+{
+  OSS oss(false);
+  oss << getClassName() << "("
+      << "covariance model=" << covarianceModel_.__str__(offset)
+      << ", basis=" << basis_.__str__(offset)
+      << ", trend coefficients=" << beta_.__str__(offset) << ")";
+  return oss;
+}
+
+/* Basis accessor */
+Basis GaussianProcessFitterResult::getBasis() const
+{
+  return basis_;
+}
+
+/* Trend coefficients accessor */
+Point GaussianProcessFitterResult::getTrendCoefficients() const
+{
+  return beta_;
+}
+
+/** Trend function accessor */
+Function GaussianProcessFitterResult::getTrendFunction() const
+{
+  return trendFunction_;
+}
+
+/* Covariance models accessor */
+CovarianceModel GaussianProcessFitterResult::getCovarianceModel() const
+{
+  return covarianceModel_;
+}
+
+/** Regression matrix accessor */
+Matrix GaussianProcessFitterResult::getRegressionMatrix() const
+{
+  return regressionMatrix_;
+}
+
+Scalar GaussianProcessFitterResult::getOptimalLogLikelihood() const
+{
+  return optimalLogLikelihood_;
+}
+
+GaussianProcessFitterResult::LinearAlgebra GaussianProcessFitterResult::getLinearAlgebraMethod() const
+{
+  return linearAlgebraMethod_;
+}
+
+/* process accessor */
+Process GaussianProcessFitterResult::getNoise() const
+{
+  // Define noise process
+  if (getCovarianceModel().getClassName() == "DiracCovarianceModel")
+  {
+    // Here it is assumed that the covariance model parameters are the
+    // marginal amplitude.
+    const Point sigma(getCovarianceModel().getParameter());
+    const CorrelationMatrix R(getCovarianceModel().getOutputCorrelation());
+    const Normal dist(Point(sigma.getSize(), 0.0), sigma, R);
+    const WhiteNoise noise(dist);
+    return noise;
+  }
+  // Other covariance models
+  const GaussianProcess noise(getCovarianceModel(), Mesh(getInputSample()));
+  return noise;
+}
+
+/* Method that returns the covariance factor - Lapack */
+TriangularMatrix GaussianProcessFitterResult::getCholeskyFactor() const
+{
+  return covarianceCholeskyFactor_;
+}
+
+/** rho accessor */
+Point GaussianProcessFitterResult::getRho() const
+{
+  return rho_;
+}
+
+void GaussianProcessFitterResult::setCholeskyFactor(const TriangularMatrix & covarianceCholeskyFactor,
```

Comment:
So we must necessarily have a HMat covariance matrix in order to set the Cholesky factor?

Reply:
Yes, like the current implementation.  An "empty one" if we are using `LAPACK`